### DISCUSSION TITLE
Food matching multiple matches

### DISCUSSION THREAD

**Discussion by rbroth** [micronutrientsupport/database-architecture#115] [General]
What happens if we have a food genus that matches multiple entries in a single Food Composition Table?

- select FCT entry at random
- use median 
- use mean
- When importing an FCT, manually make sure that there is only one entry per food genus - choose the FCT entry most appropriate to that food genus and expunge the others

**Comment by rbroth** [comment]
When a single Food Composition table contains multiple instances of the same food genus, we want to take the mean values of the FCt entries

**Reply by LuciaSegovia** [reply]
I don't fully understand this. Could you give an example? Thanks!

**Reply by rbroth** [reply]
Food Composition Table "Malawi FCT 2019" contains three entries: Beafsteak tomatoes (protein: 1g), cherry tomatoes (protein: 5g), organic salad tomatoes (protein: 2g). All of them have Food genus "57348: Tomato". What value of protein should we use when determining the nutrient intake of a Malawian who ate 100g of "tomatoes"? Answer: the mean of all "57348: Tomato" entries (2.66g protein)

**Reply by LuciaSegovia** [reply]
Ok, idealy that shouldn't happen. I mean that two food entries share the same genus in the same FCT. I need to think about it. I am not sure if that is the correct approach.

**Reply by rbroth** [reply]
For future reference, since I had to dig through the code to remember: for the time being we will use the average (mean) nutrient composition for a food_genus in an fct. In the future, FCTs will be processed so that there are no duplicate food_genus entries in an FCT



|food_genus_id|original_food_name|fct_source_id|name|count|
|-------------|------------------|-------------|----|-----|
|1323.01|Citrus, Other|24|Eastern-Africa|2|
|1323.01|Oranges, Mandarines|24|Eastern-Africa|2|
|1523.01|Fish, Liver Oil|24|Eastern-Africa|2|
|1523.01|Fish, Liver Oil|25|Middle-Africa|2|
|1323.01|Oranges, Mandarines|25|Middle-Africa|2|
|1323.01|Citrus, Other|25|Middle-Africa|2|
|1323.01|Citrus, Other|26|Southern-Africa|2|
|1323.01|Oranges, Mandarines|26|Southern-Africa|2|
|1523.01|Fish, Liver Oil|26|Southern-Africa|2|
|1323.01|Oranges, Mandarines|27|Western-Africa|2|
|1523.01|Fish, Liver Oil|27|Western-Africa|2|
|1323.01|Citrus, Other|27|Western-Africa|2|

**Reply by LuciaSegovia** [reply]
Thanks for the heads-up! I thought that was fixed already! I'll check it up :)
